docs/user-guide/CAI.rst

```diff
@@ -0,0 +1,308 @@
+
+   This will download the dataset to ``/path/to/nvidia_sft_datablend_v1_train.json``
+
+
+4. Download and process preference helpfulness dataset:
+   
+   .. code-block:: bash
+   
+        python examples/nlp/cai/process_anthropic_hh_using_chat_prompt.py \
+           --output-dir /path/to/anthropic_helpful_only \
+           --dataset-dir-name helpful-base helpful-online helpful-rejection-sampled \
+           --output-file-name-prefix anthropic_helpful_only
+
+
+Step 2: Generate and revise responses to harmful prompts creating the SL-CAI dataset
+###################################################################################################
+
+Run an inference server in the background using the following command:
+
+.. code-block:: bash
+
+   python /opt/NeMo/examples/nlp/language_modeling/megatron_gpt_eval.py \
+           gpt_model_file=/models/mistral/mistral-7b-Instruct.nemo \
+           pipeline_model_parallel_split_rank=0 \
+           server=True \
+           tensor_model_parallel_size=8 \
+           pipeline_model_parallel_size=1 \
+           trainer.precision=bf16 \
+           trainer.devices=8 \
+           trainer.num_nodes=1 \
+           web_server=False \
+           port=5999 
+
+Please wait for the server to be ready before proceeding.
+
+.. code-block:: bash
+
+   python examples/nlp/cai/generate_sl_cai_dataset.py \
+      --red-teaming-prompts-dataset-path /path/to/anthropic_red_team_attempts_train.json \
+      --few-shot-prompts-dataset-path few_shot_samples_example.json \
+      --critique-revision-instructions-path CritiqueRevisionInstructions.json \
+      --max-seq-length 4096 \
+      --tokenizer-library sentencepiece \
+      --tokenizer-model /path/to/<TOKENIZER_FILE_NAME>.model \
+      --helpfulness-dataset-path /path/to/nvidia_sft_datablend_v1_train.json \
+      --output-filepath /path/to/cai_revisions_aligner_chat_template.jsonl \
+      --port 5999 \
+      --host <hostname or IP address of the inference service>
+
+This will generate an SL-CAI dataset of prompts and revised responses as ``cai_revisions_aligner_chat_template.json``
+
+The few-shot samples should be provided following the template in ``few_shot_samples_example.json`` (filling in the `content` tags, and choosing how many samples to use), and should include a red teaming prompt, a response from the helpful model (e.g. ``Mistral-7B`` in this tutorial), critique and revision requests and responses. The Anthropic repo outlines an example `here <https://github.com/anthropics/ConstitutionalHarmlessnessPaper/blob/main/prompts/CritiqueRevisionFewShotPrompts.json>`_.
+
+*NOTE: The tokenizer file can be found by extracting the .nemo checkpoint using `tar -xf /models/mistral/mistral-7b-Instruct.nemo`.
+There are 2 tokenizer files that end with `*.model` in the model checkpoint and they are the same, so you can use either one for data processing.*
```

Comment:
Delete * after data processing.* It shows up in the rendered text. I don't think it is needed, but pls check.

Reply:
Thanks, fixed.